### PR TITLE
Add rerun option for integration tests in CI environments and bump Node version

### PR DESCRIPTION
**Pull request by gaetanmaisse**
## Issue

NA

## Description

Adjusted the maven commands in the integration tests job to include "rerunFailingTestsCount=2".
This change was necessitated due to the random failure of several integration tests related to RabbitMQ or Websocket in the CI environment. This should assist in reducing the number of false positives and avoid wasting resources on rerunning the entire suite for sporadic failures.

I also bumped up the version of Node as 16 is not supported anymore

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-rkezemvvgs.chromatic.com)

### CODE REVIEW

File: .circleci/ci/.nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-16.10.0
\ No newline at end of file
+20
```
Comment (phiz71) [inline .circleci/ci/.nvmrc:1]: I have no opinion for `20` or `20.x.y`, but maybe @Okhelifi and @ThibaudAV have one?

**Review by phiz71 — APPROVED** [review]
LGTM, just a question regarding the node version

File: .circleci/ci/.nvmrc (hunk 1 above)
Comment (Okhelifi) [reply .circleci/ci/.nvmrc:1]: I don't have a strong opinion on it 🤷🏼

Comment (ThibaudAV) [reply .circleci/ci/.nvmrc:1]: I don't think there's too much risk and it keeps us up to date. If there's a problem, we can fix the minor later 🤔

**Pull request merged**